libobs/util/platform-nix.c

```diff
@@ -458,7 +460,10 @@ struct os_dirent *os_readdir(os_dir_t *dir)
 	if (!dir->cur_dirent)
 		return NULL;
 
-	strncpy(dir->out.d_name, dir->cur_dirent->d_name, 255);
+	const size_t length = strlen(dir->cur_dirent->d_name);
```

Comment:
This changes behavior as it no longer truncates right? no clue if anyone relies on that (probably someone does) but its why we have rejected the other fixes for the warning.

Reply:
A number of stars would need to align for truncation to have the appearance of correctness: path longer than 255, valid path exactly at 255 chars, and contents that reflect the actual path in the truncated path to fool user into thinking they entered the correct path.

Folder matching by accidental 255 character truncation is bad, and I don't think we should support it.